### ISSUE TITLE
False positive for Lint/UselessAccessModifier modules extending ActiveSupport::Concern

### ISSUE THREAD

**Issue by RST-J** [rubocop-hq/rubocop#7298]
## Expected behavior
The `private` modifier in the `class_methods` block is not flagged as `Lint/UselessAccessModifier: Useless 'private' access modifier.`

## Actual behavior
The `private` modifier in the `class_methods` block is flagged as `Lint/UselessAccessModifier: Useless 'private' access modifier.`

## Steps to reproduce the problem
Run rubocop on a file with the following contents:
```ruby
# frozen_string_literal: true

module Foo
  extend ActiveSupport::Concern

  private

  def private_instance_bar
  end

  class_methods do
    private

    def private_class_bar
    end
  end
end
```

## RuboCop version
0.74.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-linux)

**Comment by Drenmi** [comment]
Is this `class_methods` convention common enough outside of Rails to warrant a special condition? I'm inclined to say yes, mostly based on the fact that I don't think there will be any false positives, i.e. `class_methods` implementations that don't honour visibility scopes.

**Comment by RST-J** [comment]
Good question. Maybe this should go into `rubocop-rails`?
I wasn't really sure whether to open this issue over there or not. I think I brought it up here because `ActiveSupport` can be used without rails if one wants to.
As I have not much experience with Ruby off Rails, I cannot tell.

**Comment by kliuless** [comment]
I ran into what seems like the same issue:
```ruby
module TestModule
  extend ActiveSupport::Concern

  included do
    private

    attr_writer :testvar1
  end

  class_methods do
    private

    attr_writer :testvar2
  end
end
```
The `private` inside `class_methods` is flagged as `Useless private access modifier`

**Comment by halilim** [comment]
Another way to reproduce:

```rb
module Foo
  extend ActiveSupport::Concern

  included do
    private

    attr_reader :bar
  end

  private

  def private_instance_method
    # ...
  end
end
```

I can put the private instance methods into `included` as a workaround but then it causes a `Metrics/BlockLength` warning after a few methods.

**Comment by kliuless** [comment]
Re: @halilim's example, sometimes instance methods should be outside of `included`, i.e. when you want them to be part of the inheritance chain. Putting them inside will add the method individually to each class or module that `include`s your module, preventing `super` from working.

**Comment by stefankroes** [comment]
This is definitely still an issue: got multiple false positives when applying this cop to our project. Seeing this with both `class_methods do` and `describe do`. I would say: when an access modifier appears in a block, you should ignore it or consider the whole block in isolation as you don't know how it will be evaluated.

**Comment by will-gant** [comment]
Yeah still an issue.

In my case, this gave a false positive:

```ruby
module SomeModule
  class << self
    private
    
    ...
  end
end
```

**Comment by ydakuka** [comment]
Please create issues if relevant.

**Comment by corsonknowles** [comment]
Just stopping by because the inaccuracy here is still an issue.
